src/llmcompressor/transformers/sparsification/sparse_model.py

```diff
@@ -153,6 +138,14 @@ def skip(*args, **kwargs):
 
         return model
 
+    # Add the from_pretrained method to the new class
+    setattr(new_class, 'from_pretrained', from_pretrained)
```

Comment:
```suggestion
    return type(
        hf_model_class.__name__,
        (hf_model_class,),
        {
            "from_pretrained": from_pretrained
        }
    )
```

Reply:
Could also move `modify_save_pretrained` logic into the `wrap_hf_model_class` function, then `{"save_pretrained": save_pretrained_compressed(model.save_pretrained)}

Reply:
Thanks for the code suggestion! I'll leave the modify_save_pretrained change out of this for now since it wasn't originally part of SparseAutoModelForCausalLM